### PR TITLE
Add database with backup restore scripts

### PR DESCRIPTION
**Pull request by sbrunner**
Backport of https://github.com/camptocamp/tadao_c2cgeoportal/pull/10/files

### CODE REVIEW

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                    pass
+
+    with open(args.file) as f:
+        subprocess.check_call(
+            [
+                "docker",
+                "run",
+                "--rm",
+                f"--env=PGHOST={env['PGHOST_SLAVE']}",
+                f"--env=PGPORT={env['PGPORT']}",
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup:62]: PGPORT_SLAVE ?

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                f"--env=PGUSER={env['PGUSER']}",
+                f"--env=PGPASSWORD={env['PGPASSWORD']}",
+                f"camptocamp/postgres:{env['POSTGRES_TAG']}",
+            ]
+            + args.arg
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup:67]: Are this args for `pg_dump`, seems they are passed to `docker run`

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            if line and line[0] != "#":
+                try:
+                    index = line.index("=")
+                    env[line[:index].strip()] = line[index + 1 :].strip()
+                except ValueError:
+                    # Ignore lines that don't have a '='
+                    pass
+
+    with open(args.file) as f:
+        subprocess.check_call(
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup:56]: `subprocess.run` is recommended when possible. Here we can use it with `check=True`

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import argparse
+import subprocess
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Backup the remote database.")
+    parser.add_argument(
+        "--env", default=".env", help="The env file to use to get the connection settings, by default .env."
+    )
+    parser.add_argument("--arg", action="append", help="Additional pg_dump argument")
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup:40]: example: --arg='--schema=main', may be used mutiple times.

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-restore [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+# either expressed or implied, of the FreeBSD Project.
+
+
+import argparse
+import subprocess
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Backup the restore database.")
+    parser.add_argument("--arg", action="append", help="Additional pg_dump argument")
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-restore:37]:
```suggestion
    parser.add_argument("--arg", action="append", help="Additional pg_restore argument, example: --arg='--schema=main', may be used multiple times.")
```

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-restore [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    parser.add_argument("file", help="The file path to restore")
+    args = parser.parse_args()
+
+    with open(args.file, "w") as f:
+        subprocess.check_call(
+            ["docker-compose", "exec", "db", "pg_restore"] + args.arg,
+            stdout=f,
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-restore:44]: stdin=f

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-restore [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Backup the restore database.")
+    parser.add_argument("--arg", action="append", help="Additional pg_dump argument")
+    parser.add_argument("file", help="The file path to restore")
+    args = parser.parse_args()
+
+    with open(args.file, "w") as f:
+        subprocess.check_call(
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-restore:42]: subprocess.run(..., check=True)

**Review by arnaud-morvan — APPROVED** [review]
May need to be tested with some pg_dump or pg_restore additional args before merge.

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup (hunk 1 above)
Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup:62]: +1 :-) Thanks

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup (hunk 2 above)
Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/scaffolds/create/scripts/db-backup:67]: Oups, it was misplaced...

**Pull request merged**